src/extension.ts

```diff
@@ -932,38 +926,29 @@ const build = () => {
         cancelToken: vscode.CancellationToken
       ) => {
         cancelToken.onCancellationRequested(() => {
-          buildManager.cancel();
+          TaskManager.cancelTasks();
+          TaskManager.disposeListeners();
+          buildTask.building(false);
         });
-        idfBuildChannel.clear();
         try {
-          await buildManager.build();
-          const projDescPath = path.join(
-            workspaceRoot.fsPath,
-            "build",
-            "project_description.json"
-          );
-          updateIdfComponentsTree(projDescPath);
-          Logger.infoNotify("Build Successfully");
-        } catch (error) {
-          if (error.message === "BUILD_TERMINATED") {
-            return Logger.warnNotify(`Build is Terminated`);
+          await buildTask.build();
+          await TaskManager.runTasks();
+          if (!cancelToken.isCancellationRequested) {
+            buildTask.building(false);
+            const projDescPath = path.join(
+              workspaceRoot.fsPath,
+              "project_description.json"
+            );
+            updateIdfComponentsTree(projDescPath);
+            TaskManager.disposeListeners();
           }
+        } catch (error) {
           if (error.message === "ALREADY_BUILDING") {
             return Logger.errorNotify("Already a build is running!", error);
           }
-          if (error.message === "BUILD_TOOL_NOT_ACCESSIBLE") {
-            return Logger.errorNotify(
-              "IDF Path or IDF Tools path is invalid or not accessible",
-              error
-            );
-          }
-          if (error.code === "ENOENT") {
-            return Logger.errorNotify(
-              `Make sure you have the build tools installed and set in $PATH`,
-              error
-            );
```

Comment:
This error code don't need to handled?

Reply:
Errors are not captured anymore. They were captured within the build child process.